plugins/modules/site_intent.py

```diff
@@ -580,6 +580,15 @@ def site_requires_update(self):
         self.log("Current Site: " + str(current_site))
         self.log("Requested Site: " + str(requested_site))
 
+        if requested_site.get('type') == "building":
+            requested_address = requested_site.get('site').get('building').get('address')
```

Comment:
Can we simply write as:
`        requested_address = requested_site['site']['building']['address']
`
Same for current address as well..